.github/workflows/main.yml

```diff
@@ -0,0 +1,120 @@
+name: CI
+
+on:
+  push:
+    branches: [main, dev, test]
+
+jobs:
+
+  # Set the name of the environment based on the branch. 
+  # In practice, this is just changing 'main' to 'prod', and leaving 'dev' and 'test' the same. 
+  # It is used to set the name of the environment used for the elastic beanstalk deploy. 
+  eb-env:
+    outputs:
+      env-name: ${{ steps.env-name.outputs.env }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Environment Name
+        id: env-name
+        run: |
+          echo "::set-output name=env::${{ fromJSON('{"main":"prod","dev":"dev","test":"test"}')[github.ref_name] }}"
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+
+      # Set up Python 3.7 environment
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.7"
+
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+      
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      
+      # Install dependencies
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      # Configure AWS Credentials
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      # Run our unit tests
+      - name: Run unit tests
+        run: |
+          python -m pytest
+
+  deploy:
+    needs:
+      - eb-env
+      - test
+
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+
+      # Set up Python 3.7 environment
```

Comment:
Is there a specific reason why this repository isn't also being updated to Python 3.9? Or just not thinking about it at the moment? Mentioning this because I saw changes in requirements.txt, so it looked like you made some updates there.

Reply:
I started briefly down that path but decided against it when it looked to be more than a quick upgrade. This will involve upgrading the elastic beanstalk platform, which is probably straightforward with a little googling. Might be a good time to take care of dependabot issues too. If you want, maybe add this as a task to the github issues. Might be an nice place to keep track of repository tasks going forward.

Reply:
This reason makes sense! I'll add it as an issue to come back to soon.